src/xray.js

```diff
@@ -1,9 +1,7 @@
 const xray = require('aws-xray-sdk-core');
 
-exports.captureWithXRay = () => {
-  if (process.env.LAMBDA_TASK_ROOT) {
-    xray.captureHTTPsGlobal(require('http'));
-    xray.captureAWS(require('aws-sdk'));
-    xray.capturePromise();
-  }
+exports.captureWithXRay = function (awsSdk) {
```

Comment:
Would it be possible to make awsSdk as optional, because I think most times default aws-sdk would be fine.

Reply:
I think that using the default is actually problematic unless we add a `peerDependencies` requirement to `package.json`. If the calling project and `lambda-tools` do not agree on the `aws-sdk` version then the caller's SDK instance will not get instrumented and there will not be any warning.